File: java/src/test/java/org/taymyr/lagom/javadsl/broker/TestTopicPublisher.java
package org.taymyr.lagom.javadsl.broker;

import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.Descriptor;
import com.lightbend.lagom.javadsl.api.Service;
import com.lightbend.lagom.javadsl.api.ServiceCall;

import static com.lightbend.lagom.javadsl.api.Service.named;
import static com.lightbend.lagom.javadsl.api.Service.pathCall;

public interface TestTopicPublisher extends Service {

    ServiceCall<NotUsed, NotUsed> publishWithoutKey(String msg);

    ServiceCall<NotUsed, NotUsed> publishWithKey(String msg);

    @Override
    default Descriptor descriptor() {
        return named("test-topic-publisher").withCalls(
            pathCall("/test-topic-publisher/without-key/:msg", this::publishWithoutKey),
            pathCall("/test-topic-publisher/with-key/:key", this::publishWithKey)
        ).withAutoAcl(true);
    }
}

File: src/test/kotlin/org/taymyr/lagom/javadsl/broker/TestTopicPublisher.kt
package org.taymyr.lagom.javadsl.broker

import akka.NotUsed
import com.lightbend.lagom.javadsl.api.Descriptor
import com.lightbend.lagom.javadsl.api.Service
import com.lightbend.lagom.javadsl.api.Service.named
import com.lightbend.lagom.javadsl.api.Service.pathCall
import com.lightbend.lagom.javadsl.api.ServiceCall
import kotlin.reflect.jvm.javaMethod

interface TestTopicPublisher : Service {
    fun publishWithoutKey(msg: String): ServiceCall<NotUsed, NotUsed>

    fun publishWithKey(msg: String): ServiceCall<NotUsed, NotUsed>

    @JvmDefault
    override fun descriptor(): Descriptor {
        return named("test-topic-publisher").withCalls(
            pathCall<NotUsed, NotUsed>("/test-topic-publisher/without-key/:msg", TestTopicPublisher::publishWithoutKey.javaMethod),
            pathCall<NotUsed, NotUsed>("/test-topic-publisher/with-key/:key", TestTopicPublisher::publishWithKey.javaMethod)
        ).withAutoAcl(true)
    }
}